addon-test-support/format-violation.ts

```diff
@@ -7,34 +7,25 @@ import type { Result } from 'axe-core';
  * @param {string | string[]} markup (optional) string of HTML relevant to the violation
  */
 export default function formatViolation(
-  violation: Partial<Result> | undefined,
-  markup?: string | string[]
+  violation: Partial<Result>,
+  markup: string[]
 ) {
-  if (!violation) {
-    throw new Error(
-      'formatViolation called without required parameter: violation'
-    );
-  }
   if (!violation.impact || !violation.help || !violation.helpUrl) {
     throw new Error(
       'formatViolation called with improper structure of parameter: violation. Required properties: impact, help, helpUrl.'
     );
   }
 
   let count = 1;
+  let formattedMarkup = '';
 
-  if (markup) {
-    if (Array.isArray(markup)) {
-      count = markup.length;
-      markup = markup.join('\n');
-    }
-    markup = ` Offending nodes are: \n${markup}`;
-  } else {
-    markup = '';
+  if (markup.length) {
```

Comment:
is it possible to have a violation without a corresponding node?

Reply:
Based on the [type definition in the `axe.Result`](https://github.com/dequelabs/axe-core/blob/42738b5258f058a36b00533e8c5cfdc0f1bcdbed/axe.d.ts#L114) type, `nodes` is always at least an empty array. When we [map over the nodes in the default reporter](https://github.com/ember-a11y/ember-a11y-testing/blob/f142821ab7f22a800eb44c4c5b36970168258146/addon-test-support/reporter.ts#L14), we always at least return an empty array, so this should be safe.